.github/workflows/test.yaml

```diff
@@ -0,0 +1,56 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Test
```

Comment:
Any reason not to name this Lint (and rename the file lint.yaml)?

Reply:
Because I'll add a job that builds and runs unit tests to this workflow later like apache/arrow-js did.
See also: https://github.com/apache/arrow-js/blob/main/.github/workflows/test.yaml

apache/arrow has a separated workflow for lint but I think that we don't need to do it in apache/arrow-swift. Because we don't need to use different `on.{push,pull_request}.paths` for lint and test in apache/arrow-swift. (We need to use different `on.{push,pull_request}.paths` for lint and test in apache/arrow because apache/arrow has multiple implementations.)

Reply:
Sounds good. Thanks for the explanation!